thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/function/RatioOutlierFunction.java

```diff
@@ -0,0 +1,202 @@
+package com.linkedin.thirdeye.detector.function;
+
+import com.linkedin.pinot.pql.parsers.utils.Pair;
+import com.linkedin.thirdeye.anomaly.views.AnomalyTimelinesView;
+import com.linkedin.thirdeye.api.DimensionMap;
+import com.linkedin.thirdeye.api.MetricTimeSeries;
+import com.linkedin.thirdeye.dashboard.views.TimeBucket;
+import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import com.linkedin.thirdeye.datalayer.dto.RawAnomalyResultDTO;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * See params for property configuration.
+ * <p/>
+ * min - lower threshold limit for average (inclusive). Will trigger alert if datapoint < min
+ * (strictly less than)
+ * <p/>
+ * max - upper threshold limit for average (inclusive). Will trigger alert if datapoint > max
+ * (strictly greater than)
+ */
+public class RatioOutlierFunction extends BaseAnomalyFunction {
+  public static final String DEFAULT_MESSAGE_TEMPLATE = "change : %.2f %%, currentVal : %.2f, min : %.2f, max : %.2f";
+  public static final String MIN_VAL = "min";
+  public static final String MAX_VAL = "max";
+
+  final static Logger LOG = LoggerFactory.getLogger(RatioOutlierFunction.class);
+
+  public static String[] getPropertyKeys() {
+    return new String [] {MIN_VAL, MAX_VAL};
+  }
+
+  @Override
+  public List<Pair<Long, Long>> getDataRangeIntervals(Long monitoringWindowStartTime,
+      Long monitoringWindowEndTime) {
+    List<Pair<Long, Long>> startEndTimeIntervals = new ArrayList<>();
+    startEndTimeIntervals.add(new Pair<>(monitoringWindowStartTime, monitoringWindowEndTime));
+    return startEndTimeIntervals;
+  }
+
+  @Override
+  public List<RawAnomalyResultDTO> analyze(DimensionMap exploredDimensions,
+      MetricTimeSeries timeSeries, DateTime windowStart, DateTime windowEnd,
+      List<MergedAnomalyResultDTO> knownAnomalies) throws Exception {
+    List<RawAnomalyResultDTO> anomalyResults = new ArrayList<>();
+    // Parse function properties
+    Properties props = getProperties();
+
+    // Get min / max props
+    Double min = null;
+    if (props.containsKey(MIN_VAL)) {
+      min = Double.valueOf(props.getProperty(MIN_VAL));
+    }
+
+    Double max = null;
+    if (props.containsKey(MAX_VAL)) {
+      max = Double.valueOf(props.getProperty(MAX_VAL));
+    }
+
+    // Metric
+    String topicMetric = getSpec().getTopicMetric();
```

Comment:
just use getSpec().getMetric().. I dont see a reason to overrider plain simple getters/setters.